### ISSUE TITLE
Controlling the main game

### ISSUE THREAD

**Issue by Pprime1** [Pprime1/csasm#20]
Need to find a way to restart the whole game (return to the opening screen form) if player chooses an invalid game code. ... ERROR-HANDLING branch refers.

**Comment by judsona** [comment]
I’ll have a look at this tomorrow.. see what I can find

**Comment by judsona** [comment]
Hey Peter,

I made a commit for this, I didn't add the display part just how it sends back from the server.. see commit for all my changes 

https://github.com/Pprime1/csasm/commit/118e89af89ca4f1f80e9eddeba3b5da6c3249db8

Added Reward Page with redirect to cancel current players session (they will leave the game)

Added game query at startup so we aren't constantly querying database for same information (must restart server on game information change... can be made a periodic update if need be)

Added an response to socket.emit('join-a-game') in the client.js so we can send an error message back on joining invalid game.. See line 100-101 in client.js

**Comment by judsona** [comment]
I used this functionality https://socket.io/docs/v4/emitting-events/#Acknowledgements to send a response back to a user attempting to join a game.

**Comment by Pprime1** [comment]
Thankyou. I think I got that working. But for some reason I cannot get the game description variable to display in index.ejs no matter how I try (I have littered the code now with failed attempts). Cannot see what I am missing there. ?

IF I can get it to show in the game display section (index.ejs) then I will need to find a way to send it and some other key game display data to the new reward.ejs file - perhaps following your localstorage approach. *TODO*

little steps forward .... slowly slowly.
